wirecard/oxid-ee
# WDPD-21: Send Fraud Prevention Data (Feedback)

### This PR

* Modifies fields sent as "additional data" when making a transaction:
  * Removes field `consumer-id` from additional data
  * Adds field `address.country` to account holder & shipping account holder
  * Adds field `address.state` to account holder & shipping account holder, if a country state is given
  * Adds field `gender` to account holder & shipping account holder
  * Adds field `date-of-birth` to account holder
  * Uses street/house number for field `address.street2`
* Adds Payment model (extended)
* Refactors checks for whether or not a payment is a Wirecard payment
* Adds a method to the Payment method model to set the custom fields to be printed in the admin config

### How to Test

* Make a PayPal transaction
* Inspect the logs (`source/log/oxideshop.log`)
* Find the "Request body" JSON and confirm that:
  * The field `consumer-id` is not present
  * The field `address` contains the properties `country` and `state` (if applicable)
  * The field `gender` is set if specified by the user
  * The field `date-of-birth` is set if specified by the user

### Jira Links

* [WDPD-21](https://jira.parkside.at/browse/WDPD-21)
